### PR TITLE
cmd/k8s-operator: sprinkle debug/info logging throughout.

### PR DESCRIPTION
**Pull request by danderson**
As is convention in the k8s world, use zap for structured logging. For development, OPERATOR_LOGGING=dev switches to a more human-readable output than JSON.

Updates #502

Signed-off-by: David Anderson <danderson@tailscale.com>

### CODE REVIEW

File: cmd/k8s-operator/operator.go [hunk 1]
```diff
@@ -347,14 +374,16 @@ func (a *ServiceReconciler) Reconcile(ctx context.Context, req reconcile.Request
 		return reconcile.Result{}, fmt.Errorf("failed to get device ID: %w", err)
 	}
 	if tsHost == "" {
+		logger.Debugf("no Tailscale hostname known yet, waiting for proxy pod to finish auth")
 		// No hostname yet. Wait for the proxy pod to auth.
 		svc.Status.LoadBalancer.Ingress = nil
 		if err := a.Status().Update(ctx, svc); err != nil {
 			return reconcile.Result{}, fmt.Errorf("failed to update service status: %w", err)
 		}
-		return reconcile.Result{RequeueAfter: 10 * time.Second}, nil
```
Comment (skriptble) [inline cmd/k8s-operator/operator.go:355]: Is this cleanup from the previous PR?

File: cmd/k8s-operator/operator_test.go [hunk 2]
```diff
@@ -54,7 +60,7 @@ func TestLoadBalancerClass(t *testing.T) {
 		},
 	})
 
-	expectRequeue(t, sr, "default", "test")
+	expectReconciled(t, sr, "default", "test")
```
Comment (skriptble) [inline cmd/k8s-operator/operator_test.go:63]: Same as above, was is this related to the logging or just a missed change from the previous PR?

**Review by skriptble — APPROVED** [review]
It looks like there are two changes unrelated to the logging, but there's only 1 commit this time. Not sure if it's worth a separate commit for those two changes though.

File: cmd/k8s-operator/operator.go (hunk 1 above)
Comment (danderson) [reply cmd/k8s-operator/operator.go:355]: Cleanup found while working through the code and adding logging. The previous code was mostly harmless, just queued some pointless no-op work for later.

File: cmd/k8s-operator/operator_test.go (hunk 2 above)
Comment (danderson) [reply cmd/k8s-operator/operator_test.go:63]: Yeah just a missed reconcile change.

File: cmd/k8s-operator/operator.go (hunk 1 above)
Comment (skriptble) [reply cmd/k8s-operator/operator.go:355]: 👍🏾

File: cmd/k8s-operator/operator_test.go (hunk 2 above)
Comment (skriptble) [reply cmd/k8s-operator/operator_test.go:63]: 👍🏾

**Comment by danderson** [comment]
It was easy enough to split up, so I made this into 2 changes: the headline logging change, and a small followup for the timed reconcile thing.

**Pull request merged**